test/run

```diff
@@ -358,6 +358,15 @@ testOldNpmWithLockfile() {
   assertCapturedSuccess
 }
 
+testWarnAwsProxy() {
+  env_dir=$(mktmpdir)
+  echo "http://localhost:5001" > $env_dir/HTTP_PROXY
+  echo "http://localhost:5001" > $env_dir/HTTPS_PROXY
+  compile "node-10" "$(mktmpdir)" $env_dir
+  assertCaptured "Your build may fail if NO_PROXY is not set to amazonaws.com"
+  assertCapturedError
+}
```

Comment:
Do you mind creating 2 tests for this that distinguish between the `HTTP_PROXY` and `HTTPS_PROXY` since people may only use 1?